README.MD

```diff
@@ -141,21 +141,23 @@ See [CLI documentation](#command-line-interface)
 
 #### Programmatic
 
-```
-import { generateRoutes, generateSwaggerSpec } from 'tsoa';
+```typescript
+import { generateRoutes, generateSwaggerSpec, SwaggerConfig } from 'tsoa';
 
 (async () => {
-    await generateRoutes({
+    const swaggerOptions: SwaggerConfig = {
         basePath: "/api",
         entryFile: "./api/server.ts",
-        routesDir: "./api",
-    });
+        outputDirectory: "./api/dist",
+    }
+
+    await generateSwaggerSpec(swaggerOptions);
 
-    await generateSwaggerSpec({
+    await generateRoutes({
         basePath: "/api",
         entryFile: "./api/server.ts",
-        outputDirectory: "./api/dist",
-    });
+        routesDir: "./api",
+    }, swaggerOptions);
```

Comment:
I had to update the documentation sincee the new option `noImplicitAdditionalProperties` is required by both cli commands (`generateRoutes` and `generateSwaggerSpec`) due to it affecting both swagger creation and runtime validation.